server/src/main/java/org/elasticsearch/indices/TimestampFieldMapperService.java

```diff
@@ -104,7 +106,15 @@ public void applyClusterState(ClusterChangedEvent event) {
 
             if (hasUsefulTimestampField(indexMetadata) && fieldTypesByIndex.containsKey(index) == false) {
                 logger.trace("computing timestamp mapping for {}", index);
-                final PlainActionFuture<DateFieldMapper.DateFieldType> future = new PlainActionFuture<>();
+                final PlainActionFuture<DateFieldMapper.DateFieldType> future = new PlainActionFuture<>() {
+                    @Override
+                    protected boolean blockingAllowed() {
```

Comment:
We can remove this override now that we're setting the timeout.

Reply:
Oh, I thought I removed it... I need more ☕  today